akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala

```diff
@@ -125,10 +125,10 @@ private[http2] object PersistentConnection {
           requestOut.fail(new RuntimeException("connection broken"))
 
           if (connectsLeft.contains(0)) {
-            failStage(cause)
+            failStage(new RuntimeException(s"Connection failed after $maxAttempts attempts"))
```

Comment:
```suggestion
            failStage(new RuntimeException(s"Connection failed after $maxAttempts attempts"), cause)
```

Keep the cause around? Or maybe add the `cause.getMessage` in a log? (like in line 131)